src/dhcp_proxy/dhcp_service.rs

```diff
@@ -49,6 +49,25 @@ impl DhcpV4Service {
     pub fn new(nc: NetworkConfig, timeout: u32) -> Result<Self, DhcpServiceError> {
         let mut config = DhcpV4Config::new_proxy(&nc.host_iface, &nc.container_mac_addr);
         config.set_timeout(timeout);
+
+        if !nc.host_name.is_empty() {
+            let cid: String = nc.container_id.chars().take(16).collect();
```

Comment:
seems like there is no reason to limit to 16 here, more chars the less likely we have a conflict with other systems on the network. It seems like the client ID filed can be up to 63 chars?

Reply:
I limited it to 16 because that's the size of a UUID which is a common client id.  No reason not to expand it though.  Like every one of the hundreds of RFCs I've read over the years, [RFC2132 section 9.14](https://datatracker.ietf.org/doc/html/rfc2132#section-9.14) doesn't tell you what you _really_ want to know.  It says the minimum length is 2 but it doesn't specify a maximum length.  It does mention using an FQDN as an example and since a FQDN can be 253 bytes, a 64 byte container ID should be fine.